main_test.go

```diff
@@ -10,11 +10,45 @@ import (
 func TestTimeGranularity(t *testing.T) {
 	baseTime := time.Date(2017, 7, 11, 12, 0, 0, 0, time.UTC)
 
-	start, end := startEndFromGranularity(baseTime, "day")
+	start, end := startEndFromGranularity(baseTime, "day", false)
 	assert.Equal(t, start, time.Date(2017, 7, 11, 0, 0, 0, 0, time.UTC))
 	assert.Equal(t, end, time.Date(2017, 7, 12, 0, 0, 0, 0, time.UTC))
 
-	start, end = startEndFromGranularity(baseTime, "week")
+	start, end = startEndFromGranularity(baseTime, "hour", false)
+	assert.Equal(t, start, time.Date(2017, 7, 11, 12, 0, 0, 0, time.UTC))
+	assert.Equal(t, end, time.Date(2017, 7, 11, 13, 0, 0, 0, time.UTC))
+
+	// Simulate timestamps that cross timezones in PT vs UTC
+	baseTime = time.Date(2017, 7, 11, 4, 0, 0, 0, time.UTC)
+
+	start, end = startEndFromGranularity(baseTime, "day", false)
+	assert.Equal(t, start, time.Date(2017, 7, 11, 0, 0, 0, 0, time.UTC))
+	assert.Equal(t, end, time.Date(2017, 7, 12, 0, 0, 0, 0, time.UTC))
+
+	start, end = startEndFromGranularity(baseTime, "day", true)
 	assert.Equal(t, start, time.Date(2017, 7, 10, 0, 0, 0, 0, time.UTC))
```

Comment:
hmm, if we're getting a startEndFromGranularity for a day for PT, shouldn't this give us the PT beginning and ending of the day?
I'd expect `time.Date(2017, 7, 10, 7, 0, 0, 0, time.UTC)` and `time.Date(2017, 7, 11, 16, 0, 0, 0, time.UTC))`, if I'm reading right.

Likely you'll have to do something a bit ugly, see:
- https://stackoverflow.com/questions/36988681/golang-time-time-round-to-day
- https://stackoverflow.com/questions/25254443/return-local-beginning-of-day-time-object-in-go